api/src/main/java/org/openmrs/module/muzimafingerPrint/api/impl/MuzimafingerPrintServiceImpl.java

```diff
@@ -85,7 +87,36 @@ public PatientFingerPrintModel identifyPatient(String fingerprint) throws IOExce
         }
         if(patient == null)
             return null;
-        return new PatientFingerPrintModel(patient.getUuid(),fingerprint, patient.getId(), patient.getGivenName(), patient.getFamilyName(), patient.getGender());
+        else {
+            mfingerprint = getFingerprintByPatientUUID(patient.getUuid());
+            if (mfingerprint != null) {
+                patfingerprint = mfingerprint.getFingerprint();
+            }
+            return new PatientFingerPrintModel(patient.getUuid(), patfingerprint, patient.getId(), patient.getGivenName(), patient.getFamilyName(), patient.getGender());
```

Comment:
Is there a reason that patient fingerprint model contains duplicate information of the patient? Why not use composition and add patient inside the model?